src/prefect/orion/schemas/core.py

```diff
@@ -996,6 +996,34 @@ def helpful_error_for_missing_default_queue_id(cls, v):
             )
         return v
 
+    @validator("base_job_template")
+    def validate_base_job_template(cls, v):
+        if v == dict():
+            return v
+
+        job_config = v.get("job_configuration")
+        variables = v.get("variables")
+        if not (job_config and variables):
+            raise ValueError(
+                "The `base_job_template` must contain both a `job_configuration` key"
+                " and a `variables` key."
+            )
+        template_variables = set()
+        for template in job_config.values():
+            template_variables.add(
+                # {{ var }} -> var
+                template.strip("{} ")
```

Comment:
This feels a bit hacky. This is my first time with jinja, so if there's a cleaner way to do this, I'm all ears.

Reply:
You could use regex: `re.findall("\{\{\s(.*?)\s\}\}", template)`. That may not work in all scenarios, but I think it will be more robust than `strip`.

Reply:
Solid - going with `r"{{\s*(.*?)\s*}}"` so that I can deal with people messing with the whitespace on the template too.